### ISSUE TITLE
Support specifiers like %f in RequiresMountsFor in [Unit] section

### ISSUE THREAD

**Issue by JohnstonJ** [systemd/systemd#2829]
### Submission type
- [ ] Bug report
- [X] Request for enhancement (RFE)

_NOTE: Do not submit anything other than bug reports or RFEs via the issue tracker!_
### systemd version the issue has been seen with

> 225
> 
> Also: examination of `src/core/load-fragment.c` suggest the feature request is valid even in most recent master.

_NOTE: Do not submit bug reports about anything but the two most recently released systemd versions upstream!_
### Used distribution

> Ubuntu Server 15.10
### Feature request

The `RequiresMountsFor` configuration option in the `[Unit]` section does not appear to support specifiers such as `%f`.  I briefly examined the `config_parse_unit_requires_mounts_for` function in `src/core/load-fragment.c` in latest master (commit ed0b16e2792e) and it looks like there just isn't any support for specifiers there at all (that said, this is my first time looking at systemd source code).

For example, suppose I have `LoopSetup@.service`:

```
[Unit]
RequiresMountsFor=%f
<snip>

[Service]
ExecStart=/sbin/losetup -f %f
<snip>
```

I would then like to be able to set up the loop device by: `systemctl start LoopSetup@root-swap`

The motivation here is to set up a swap file on btrfs, which requires setting up a loop device (and specifying `loop` as a mount option in `fstab` does not appear to work.  I found lots of examples on the Internet with this mount option for swap files, so maybe it is not supposed to set up a loop device in general - but that breaks btrfs).

Rather than make a separate `.service` file for every single item, I wanted to just have a single service template and use the specifiers - but it looks like the support in systemd to do that is just not there.

**Comment by poettering** [comment]
Sounds like a reasonable request.

**Comment by poettering** [comment]
Fixed by #4835
